_posts/2015-11-16-behat-to-the-fixture.md

```diff
@@ -0,0 +1,272 @@
+## Manage data bases before to load fixtures
+
+{% highlight php startinline=true %}
+class FixturesContext implements Context
+{
+    /** @BeforeSuite */
+    public static function before($event)
+    {
+        $kernel = new AppKernel("dev", true);
+        $kernel->boot();
+    
+        FixturesContext::checkDatabasesHosts($kernel);
+    
+        $application = new Application($kernel);
+        $application->setAutoExit(false);
+        FixturesContext::runConsole($application, "doctrine:schema:drop", ["--force" => true, "--full-database" => true]);
+        FixturesContext::runConsole($application, "doctrine:schema:create");
+        $kernel->shutdown();
+    }
+    
+    private static function runConsole($application, $command, $options = array())
+    {
+        $options["-e"] = "dev";
+        $options["-q"] = null;
+        $options = array_merge($options, array('command' => $command));
+        return $application->run(new ArrayInput($options));
+    }
+}
+
+{% endhighlight %}
+
+First, check database host is *localhost* (on a vagrant for example), to avoid to clear wrong databases.
+Next, delete database schema, then re-create the schema from doctrine entities. You can also use a migration script like DoctrineMigration.
+
+## Write fixtures
+
+Fixtures format is Gherkin language, as behat scenarios. Example :
+
+{% highlight gherkin %}
+Feature: fixtures library
+
+    Scenario: fixture writers
+        Given writers :
+            | name             | birth date |
+            | JK Rowling       | 1965-07-31 |
+            | Dan Brown        |            |
+            | J. R. R. Tolkien | 1892-01-03 |
+
+    Scenario: fixture books
+        Given literary genres :
+            | name            |
+            | science fiction |
+            | comedy          |
+        Given books :
+            | title                 | author     | literary genres | publication year |
+            | The Lord of the Rings | JK Rowling | science fiction | 1954             |
+            | Da Vinci Code         | Dan Brown  |                 | 2003             |
+            | Angels and Demons     | Dan Brown  |                 | 2000             |
+{% endhighlight %}
+
+Put your fixtures files **.feature* on the folder defined in *behat_fixtures.yml* (here *src/AppBundle/DataFixtures/*).
+Behat process files in alphabetical order. If you prefix the name of fixtures with a number, you can control the execution order.
+
+## Load fixtures
+
+To load fixtures, execute the script below in your root's project, and wait some seconds !
+
+{% highlight sh %}
+bin/behat --config behat_fixtures.yml
+{% endhighlight %}
+
+
+***
+
+## To conclude
+
+There are many benefits to using behat 3 to load fixtures :
+
+* Fixtures are more readable and easy to write
+* If you already use behat 3, your current steps definitions can be reused
+* If you add a column on one of your table, you have to modify only one piece of code to make evolve behat tests and fixtures
```

Comment:
`you have` => `you need`
`to make evolve` => `to modify` / `to change` (to make evolve / to evolve is not correct, there are not equivalents to "faire évoluer" that would have exactly the same meaning)